pkg/services/ngalert/provisioning/persist.go

```diff
@@ -45,3 +48,12 @@ type RuleStore interface {
 type QuotaChecker interface {
 	CheckQuotaReached(ctx context.Context, target string, scopeParams *quota.ScopeParameters) (bool, error)
 }
+
+// PersistConfig validates to config before eventually persisting it if no error occurs
+func PersistConfig(ctx context.Context, store AMConfigStore, cmd *models.SaveAlertmanagerConfigurationCmd) error {
+	cfg := &definitions.PostableUserConfig{}
+	if err := json.Unmarshal([]byte(cmd.AlertmanagerConfiguration), cfg); err != nil {
```

Comment:
I see the code just parses the request body into a `PostableUserConfig`. This would ensure we get a valid JSON from the `serializeAlertmanagerConfig()` function, but is this enough for ensuring that a valid AM configuration is saved? Do you think it would help to have a few tests?

Reply:
Update: I just saw each piece of the `PostableUserConfig` struct has custom `UnmarshalJSON` methods, which include validation steps. I'm approving these changes.

Reply:
Exactly, by calling Unmarshal you basically trigger all the validation.